File: for_atcoder/abc_154/d_Dice_in_Line.py
raw = input()
N, K = raw.split(' ')
N = int(N)
K = int(K)
str_li = input().split(' ')
num_li = list(map(int, str_li))
sum_li = [0]
for i in num_li:
    sum_li.append(sum_li[-1] + i + 1)
ans = 0
for index in range(K, N+1):
    ans_tmp = sum_li[index] - sum_li[index - K]
    if ans_tmp > ans:
        ans = ans_tmp
print(ans/2)

File: for_atcoder/abc_154/d_Dice_in_Line.cpp
#include <algorithm>
#include <fstream>
#include <iostream>
#include <iterator>
#include <ostream>
#include <string>
#include <vector>
#include <numeric>

int main(int argc, char *argv[])
{
    int N, K;
    std::cin >> N >> K;
    std::vector<int> num_li(K-1, 0);
    std::vector<int> sum_li;
    for(int i = 0; i < N; ++i){
        int tmp;
        std::cin >> tmp;
        num_li.push_back(tmp+1);
        auto start = std::next(num_li.begin(), i);
        auto end = std::next(start, K);
        int sum = std::accumulate(start, end, 0);
        sum_li.push_back(sum);
    }
    std::sort(sum_li.begin(), sum_li.end());
    double ans = sum_li.back() / 2.0;
    std::cout << ans << std::endl;
    return 0;
}
